dev/infra/Dockerfile

```diff
@@ -32,7 +32,7 @@ RUN $APT_INSTALL software-properties-common git libxml2-dev pkg-config curl wget
 RUN update-alternatives --set java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
 
 RUN curl -sS https://bootstrap.pypa.io/get-pip.py | python3.9
-RUN python3.9 -m pip install numpy pyarrow 'pandas<=1.5.1' scipy unittest-xml-reporting plotly>=4.8 sklearn 'mlflow>=1.0' coverage matplotlib openpyxl
+RUN python3.9 -m pip install numpy pyarrow 'pandas<=1.5.1' scipy unittest-xml-reporting plotly>=4.8 sklearn 'mlflow>=1.0' coverage matplotlib openpyxl 'memory-profiler==0.60.0'
 
 RUN add-apt-repository ppa:pypy/ppa
```

Comment:
```
 > [ 8/26] RUN add-apt-repository ppa:pypy/ppa:
#14 523.3 Cannot add PPA: 'ppa:~pypy/ubuntu/ppa'.
#14 523.3 ERROR: '~pypy' user or team does not exist.
------
Dockerfile:37
--------------------
  35 |     RUN python3.9 -m pip install numpy pyarrow 'pandas<=1.5.1' scipy unittest-xml-reporting plotly>=4.8 sklearn 'mlflow>=1.0' coverage matplotlib openpyxl 'memory-profiler==0.60.0'
  36 |     
  37 | >>> RUN add-apt-repository ppa:pypy/ppa
  38 |     RUN apt update
  39 |     RUN $APT_INSTALL gfortran libopenblas-dev liblapack-dev
```

@Yikun do you have an idea why this fails?

Reply:
This should be the flaky error when adding apt repository (I haven't saw this error before), I think retry can help.

Reply:
Thanks! I'll retry.